tests/test_snapshots/test_chain_snapshots.py

```diff
@@ -0,0 +1,277 @@
+        '0x05E41229Efca125057f4D96007Dc477312dB8feB': 1.04533617521779346,
+    }
+}
+
+BADGER_PRICE = 31.205460272851376
+YEARN_WBTC_PRICE = 57784.88168712999
+
+
+@pytest.fixture
+def mock_fetch_ch_balances(mocker):
+    return mocker.patch(
+        "rewards.snapshot.chain_snapshot.fetch_chain_balances",
+        return_value=BALANCES_DATA
+    )
+
+
+@pytest.fixture
+def mock_fetch_sett_balances(mocker):
+    return mocker.patch(
+        "rewards.snapshot.chain_snapshot.fetch_sett_balances",
+        return_value={
+            '0x0000000000007F150Bd6f54c40A34d7C3d5e9f56': 0.04533617521779346,
+        }
+    )
+
+
+@pytest.mark.parametrize(
+    "chain",
+    [Network.Ethereum, Network.Arbitrum]
+)
+@responses.activate
+def test_chain_snapshot__happy(mock_fetch_ch_balances, chain):
+    responses.add(
+        responses.GET, f"{badger_api}/tokens?chain={chain}",
+        json={'name': 'bBadger'},
+        status=200
+    )
+    responses.add_passthru('https://')
+    snapshot = chain_snapshot(chain, 123123)
+    native = snapshot[BADGER_TOKEN_ADDR]
+    assert native.type == BalanceType.Native
+    assert native.ratio == 1
+    assert native.token == BADGER_TOKEN_ADDR
+    assert list(native.balances.values())[0] == approx(Decimal(
+        list(BALANCES_DATA[BADGER_TOKEN_ADDR].values())[0]
+    ))
+
+    non_native = snapshot[YEARN_WBTC_ADDR]
+    assert non_native.ratio == 1
+    assert non_native.token == YEARN_WBTC_ADDR
+    assert list(non_native.balances.values())[0] == approx(Decimal(
+        list(BALANCES_DATA[YEARN_WBTC_ADDR].values())[0]
+    ))
+
+
+@pytest.mark.parametrize(
+    "chain",
+    [Network.Ethereum, Network.Arbitrum]
+)
+def test_chain_snapshot__empty(mocker, chain):
+    mocker.patch(
+        "rewards.snapshot.chain_snapshot.fetch_chain_balances",
+        return_value={}
+    )
+    snapshot = chain_snapshot(chain, 123123)
+    assert snapshot == {}
+
+
+@pytest.mark.parametrize(
+    "chain",
+    [Network.Ethereum, Network.Arbitrum]
+)
+def test_chain_snapshot__raises(mocker, chain):
+    mocker.patch(
+        "rewards.snapshot.chain_snapshot.fetch_chain_balances",
+        side_effect=Exception,
+    )
+    with pytest.raises(Exception):
+        chain_snapshot(chain, 123123)
+
+
+@pytest.mark.parametrize(
+    "chain",
+    [Network.Ethereum, Network.Arbitrum]
+)
+def test_parse_sett_balances(chain):
+    snapshot = parse_sett_balances(
+        BADGER_TOKEN_ADDR,
+        balances={
+            '0x0000000000007F150Bd6f54c40A34d7C3d5e9f56': 0.04533617521779346,
+        },
+        chain=chain,
+    )
+    assert snapshot.type == BalanceType.Native
+    assert snapshot.ratio == 1
+    assert snapshot.token == BADGER_TOKEN_ADDR
+    assert list(snapshot.balances.values())[0] == approx(Decimal(
+        list(BALANCES_DATA[BADGER_TOKEN_ADDR].values())[0]
+    ))
+
+
+@pytest.mark.parametrize(
+    "chain",
+    [Network.Ethereum, Network.Arbitrum]
+)
+def test_parse_sett_balances__blacklisted(chain, mocker):
```

Comment:
Unclear of what's being tested here.  

Blacklists apply to users rather than rewards - for example the Dev Multisig has deposits in Badger Vaults, but shouldn't receive BADGER in return, only the underlying assets.  Something more appropriate would be to pull an address from the `EMISSIONS_BLACKLIST` and ensure that no BADGER rewards are assigned to it but it is receiving things like `bcvxCRV`, and then a separate test to pull from the `REWARDS_BLACKLIST` and ensure that no rewards at all are allocated.

Reply:
It's not testing any business logic on-chain, more like checking that address gets filtered and no balance is set for the blacklisted address

Reply:
> 

I think what you are suggesting is more of a integration test, that will be checking real stuff from the chain

Reply:
And to be precise, this test is to cover this condition:
```python
    if blacklist:
        addresses_to_blacklist = {**REWARDS_BLACKLIST, **EMISSIONS_BLACKLIST}
    else:
        addresses_to_blacklist = REWARDS_BLACKLIST
```
but not with real data, but with mocked

Reply:
Yes - that would be more of an integration test but we can mock chain data.  I think I understand more now, you're adding the 0x000 to the `REWARDS_BLACKLIST` and ensuring they're removed - makes sense.